docker/tarfile/dest.go

```diff
@@ -99,7 +99,7 @@ func (d *Destination) PutBlob(ctx context.Context, stream io.Reader, inputInfo t
 	// When the layer is decompressed, we also have to generate the digest on uncompressed datas.
 	if inputInfo.Size == -1 || inputInfo.Digest.String() == "" {
 		logrus.Debugf("docker tarfile: input with unknown size, streaming to disk first ...")
-		streamCopy, err := ioutil.TempFile(tmpdir.TemporaryDirectoryForBigFiles(), "docker-tarfile-blob")
+		streamCopy, err := ioutil.TempFile(tmpdir.GetTemporaryDirectoryForBigFiles(), "docker-tarfile-blob")
```

Comment:
This also needs to be enhanced to be overridable, probably by passing a `*types.SystemContext` to `NewDestination`.

… except that it would be an API break.